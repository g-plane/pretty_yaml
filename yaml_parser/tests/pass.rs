use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
