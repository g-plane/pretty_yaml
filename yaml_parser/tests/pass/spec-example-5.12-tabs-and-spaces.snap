---
source: yaml_parser/tests/pass.rs
---
ROOT@0..96
  COMMENT@0..17 "# Tabs and spaces"
  WHITESPACE@17..18 "\n"
  DOCUMENT@18..95
    BLOCK@18..95
      BLOCK_MAP@18..95
        BLOCK_MAP_ENTRY@18..36
          BLOCK_MAP_KEY@18..24
            FLOW@18..24
              PLAIN_SCALAR@18..24 "quoted"
          COLON@24..25 ":"
          WHITESPACE@25..26 " "
          BLOCK_MAP_VALUE@26..36
            FLOW@26..36
              DOUBLE_QUOTED_SCALAR@26..36 "\"Quoted \t\""
        WHITESPACE@36..37 "\n"
        BLOCK_MAP_ENTRY@37..95
          BLOCK_MAP_KEY@37..42
            FLOW@37..42
              PLAIN_SCALAR@37..42 "block"
          COLON@42..43 ":"
          WHITESPACE@43..44 "\t"
          BLOCK_MAP_VALUE@44..95
            BLOCK@44..95
              BLOCK_SCALAR@44..95
                BAR@44..45 "|"
                BLOCK_SCALAR_TEXT@45..95 "\n  void main() {\n  \tp ..."
  WHITESPACE@95..96 "\n"
